### PR TITLE
Run pyflakes in CI

### LINKED ISSUE #47: Run pyflakes in CI

### PR DESCRIPTION
**Pull request by saraedum**
fixes #47

Pyflakes mostly reports unused variables/imports and use of undefined symbols. The latter are almost always bugs.

### CODE REVIEW

File: mclf/berkovich/berkovich_line.py [hunk 1]
```diff
@@ -1251,7 +1248,7 @@ def point_in_between(self, xi1):
             s2 = s0 + 1
         else:
             s2 = (s0+s1)/2
-        xi2 = X.point_from_discoid(phi, s2, in_unit_disk)
+        xi2 = xi1._X.point_from_discoid(phi, s2, in_unit_disk)
```
Comment (saraedum) [inline mclf/berkovich/berkovich_line.py]: @swewers What is supposed to happen here?

**Comment by saraedum** [comment]
@swewers Please have a look at my changes. Some of the might hint at actual bugs in the implementation.

File: mclf/berkovich/berkovich_line.py [hunk 2]
```diff
@@ -1278,25 +1275,23 @@ def normalized_reduction(v, f):
...
-    R = F._ring
     x = F.gen()
     r = v(f)
     m = abs(r.denominator())
     v1 = v._base_valuation
     if hasattr(v1, 'equivalence_unit'):
         fl = v.reduce(f^m*F(v1.equivalence_unit(-m*r))).factor()
         if len(fl)>0:
-            return fl[0][0]^sign(fl[0][1])
+            return fl[0][0]^sgn(fl[0][1])
```
Comment (saraedum) [inline mclf/berkovich/berkovich_line.py]: There are a lot of `^` in this area that should probably be `**`?

File: mclf/berkovich/berkovich_line.py (hunk 1 above)
Comment (swewers) [reply mclf/berkovich/berkovich_line.py]: Yes, `X` should be replaced by `xi1._X` (or `xi0._X`)
It is strange that this has never caused an error. The function `point_in_between` is used only one time, by `union` in `affinoid_domains`, but it seems that the case where it is used almost never occurs. 
Clearly there need to be more doctests!

File: mclf/berkovich/berkovich_line.py (hunk 2 above)
Comment (swewers) [reply mclf/berkovich/berkovich_line.py]: Yes! For a long time I was completely ignorant of the ^-issue in python, and I still do this wrong from time to time.

**Pull request merged**